tools/sof-dump-status.py

```diff
@@ -527,6 +598,11 @@ def dump_dapm(dapm, filter = "all"):
     # key is used. Here we output firmware path according to kernel's
     # match table.
     if ret_args.get('fwpath') is True:
+        if os.path.isdir("/lib/firmware/imx/sof"):
+            # we're currently on i.MX, just print the path
```

Comment:
This looks like... a bug?
https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/tree/imx

Reply:
DT doesn't specify the path to my knowledge, just the platform and the driver hardcodes the path. So for now this is probably correct. See for example the following excerpt from the Linux kernel code:

```
static struct sof_dev_desc sof_of_imx8qxp_desc = {
        .default_fw_path = "imx/sof",
        .default_tplg_path = "imx/sof-tplg",
        .default_fw_filename = "sof-imx8x.ri",
        .nocodec_tplg_filename = "sof-imx8-nocodec.tplg",
        .ops = &sof_imx8x_ops,
};

static struct sof_dev_desc sof_of_imx8qm_desc = {
        .default_fw_path = "imx/sof",
        .default_tplg_path = "imx/sof-tplg",
        .default_fw_filename = "sof-imx8.ri",
        .nocodec_tplg_filename = "sof-imx8-nocodec.tplg",
        .ops = &sof_imx8_ops,
};

static const struct of_device_id sof_of_imx8_ids[] = {
        { .compatible = "fsl,imx8qxp-dsp", .data = &sof_of_imx8qxp_desc},
        { .compatible = "fsl,imx8qm-dsp", .data = &sof_of_imx8qm_desc},
        { }
};
MODULE_DEVICE_TABLE(of, sof_of_imx8_ids);
```

Reply:
I'm afraid we're talking across each other, sorry for the confusion.

I'm not questioning hardcoding this path. By nature this script duplicates and hardcodes a lot of stuff from the Linux kernel and that's OK, there isn't any easy way to automatically extract this information from the kernel sources and most of it is probably part of the "kernel ABI" or duplication hardware constants in the first place so no risk of diverging.

What I find brittle is that a simple `mkdir /lib/firmware/imx/sof` on a non-IMX system can break this script.

Above there is another `# check to see if we're on i.MX` decision made and it does not look at this path at all - so why two different ways? The other code looks at DT data instead, which seems MUCH more reliable.